NBitcoin/BIP158/GolombRiceFilter.cs

```diff
@@ -481,7 +483,7 @@ public GolombRiceFilter Build()
 			return new GolombRiceFilter(filterData, n, _p, _m);
 		}
 
-		private static byte[] Compress(List<ulong> values, byte P)
+		private static byte[] Compress(ulong[] values, byte P)
```

Comment:
here you could use IEnumerable?

Reply:
meh, it is private method only used here, and a performance sensitive part. Not really worth it.